path_pattern/pattern.go

```diff
@@ -23,33 +24,45 @@ func (p *Pattern) UnmarshalText(data []byte) error {
 	return nil
 }
 
-// Match happens if the prefix of the input matches the pattern.
+func (p Pattern) regexp() *regexp.Regexp {
+	var pieces []string
+	for _, piece := range p {
+		pieces = append(pieces, piece.Regexp())
+	}
+	return regexp.MustCompile("^" + strings.Join(pieces, "/") + "$")
+}
+
+// Match happens if the pattern exactly matches the string.
 func (p Pattern) Match(v string) bool {
-	parts := strings.Split(v, "/")
+	r := p.regexp()
```

Comment:
I think we saw on the CLI that recompiling the regexp each time can be a lot of load; can we amortize it across multiple path matches?

Reply:
Good idea, fixed in 591c5e0.